src/System/Hooks/useCollectorSignals.tsx

```diff
@@ -0,0 +1,183 @@
+})
+
+const providerFactory = (signals: UseCollectorSignalsResult) => {
+  return ({ children }) => (
+    <CollectorSignalsContext.Provider value={{ signals }}>
+      {children}
+    </CollectorSignalsContext.Provider>
+  )
+}
+
+/**
+ * Hook to access the collector signals context. Must be used below
+ * the provider returned by `useLoadCollectorSignals()`
+ */
+export const useCollectorSignals = (artworkID: string): SignalResult => {
+  const signalsMap = useContext(CollectorSignalsContext)
+  console.log("***", { signalsMap, artworkID })
+  return signalsMap.signals[artworkID] ?? {}
+}
+
+/**
+ * Hook to load and process collector signals for a list of artworks
+ */
+export const useLoadCollectorSignals = ({
+  artworks,
+}: {
+  artworks?: useCollectorSignals_artworksConnection$key | null
+}): {
+  signals: UseCollectorSignalsResult
+  CollectorSignalsProvider: React.FC
+} => {
+  const { me } = useGlobalMe()
+  const meData = useFragment<useCollectorSignals_me$key>(ME_FRAGMENT, me)
+
+  const artworksData = useFragment(ARTWORKS_CONNECTION_FRAGMENT, artworks)
+  const partnerOffersEnabled = useFeatureFlag("emerald_signals-partner-offers")
+
+  const artworksDataNodes = extractNodes(artworksData)
+
+  const artworkIDs = JSON.stringify(
+    artworksDataNodes.map(artwork => artwork.internalID)
+  )
+
+  useEffect(() => {
+    console.log("*** First render of useLoadCollectorSignals")
+  }, [])
+
+  const signals = useMemo(() => {
+    const result = artworksDataNodes.reduce((acc, artworkData) => {
+      acc[artworkData.internalID] = processCollectorSignalsOnArtwork({
+        me: meData,
+        artwork: artworkData,
+        partnerOffersEnabled,
+      })
+
+      return acc
+    }, {})
+    console.log(
+      `*** processed signals for ${artworksDataNodes.length} artworks`,
+      result,
+      meData
+    )
+    return result
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [artworkIDs, meData, partnerOffersEnabled])
+
+  const signalString = JSON.stringify(signals)
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const CollectorSignalsProvider = useMemo(() => providerFactory(signals), [
+    signalString,
+  ])
+
+  return { signals, CollectorSignalsProvider: CollectorSignalsProvider }
+}
+
+/**
+ * Hook to load and process collector signals for a single artwork
+ */
+export const useCollectorSignalsOnArtwork = ({
+  artwork,
+}: {
+  artwork: useCollectorSignals_artwork$key
+}): SignalResult => {
+  const { me } = useGlobalMe()
+  const meData = useFragment<useCollectorSignals_me$key>(ME_FRAGMENT, me)
+  const partnerOffersEnabled = useFeatureFlag("emerald_signals-partner-offers")
+
+  const artworkData = useFragment(ARTWORK_FRAGMENT, artwork)
+
+  const result = processCollectorSignalsOnArtwork({
+    me: meData,
+    artwork: artworkData,
+    partnerOffersEnabled,
+  })
+
+  return result
+}
+
+const processCollectorSignalsOnArtwork = ({
+  me,
+  artwork,
+  partnerOffersEnabled,
+}: {
```

Comment:
extracted rather than calling the single artwork hook from the first because linter didn't want me iterating with a hook.